### ISSUE TITLE
/proc/sys is read-only in Docker 1.11.0

### ISSUE THREAD

**Issue by errordeveloper** [weaveworks/weave#2110]
_see https://github.com/docker/docker/issues/21334_

**Comment by errordeveloper** [comment]
I've been able to fix the issue with:
```diff
diff --git a/weave b/weave
index 60b5c07..c84f756 100755
--- a/weave
+++ b/weave
@@ -423,6 +423,7 @@ delete_iptables_rule() {
 # makes containers react more quickly to a change in the MAC address
 # associated with an IP address.
 configure_arp_cache() {
+    findmnt  /proc/sys --options rw >/dev/null 2>&1 || mount --options remount,rw /proc/sys
     $2 sh -c "echo 5 >/proc/sys/net/ipv4/neigh/$1/base_reachable_time &&
               echo 2 >/proc/sys/net/ipv4/neigh/$1/delay_first_probe_time &&
               echo 1 >/proc/sys/net/ipv4/neigh/$1/ucast_solicit"
```

**Comment by rade** [comment]
That Docker change breaks backward compatibility: existing container images with code similar to ours will stop working, and, furthermore, there doesn't appear to be a `docker run` option to get back the old behaviour, so the only possible fix is to publish modified images.

We need to report this to docker, with a simple example involving only standard images, e.g. alpine.

**Comment by errordeveloper** [comment]
https://github.com/docker/docker/issues/21649

**Comment by rade** [comment]
Fixed in Docker 1.11.0-rc4.
